### PR TITLE
[XLA:GPU} Add lowering rule from `dynamic_address_computation` fusion to command buffer `DynamicSliceFusionCmd` 

### PR DESCRIPTION
**Pull request by shawnwang18**
This PR adds lowering rules for dynamic_address_computation custom fusion to DynamicSliceFusion Cmd of command buffer.  Currently it only supports the case when the Slice offset value is constant value.

### CODE REVIEW

File: xla/service/gpu/transforms/command_buffer_scheduling.cc [hunk 1]
```diff
@@ -244,7 +243,8 @@ static bool IsCommand(const HloInstruction* hlo,
       return config.enabled_commands.contains(DebugOptions::CUDNN);
     }
     const auto& custom_config = backend_config.custom_fusion_config();
-    if (custom_config.name() == "address_computation") {
+    if ((custom_config.name() == "address_computation") ||
```
Comment (ezhulenev) [inline xla/service/gpu/transforms/command_buffer_scheduling.cc:246]: is it still adddress computation? we were replacing it with slice/dynamic_slice, but might forgot a few places

File: xla/service/gpu/runtime/command_buffer_cmd_emitter.cc [hunk 2]
```diff
@@ -191,6 +191,26 @@ static absl::StatusOr<Command> Convert(const NcclCollectiveDoneThunk& thunk) {
                                       thunk.nccl_execution_stream_id());
 }
 
+static absl::StatusOr<Command> Convert(const DynamicSliceThunk& thunk) {
+  auto cmd_sequence = std::make_unique<CommandBufferCmdSequence>();
+  auto embed_thunk = thunk.get_embeded_thunk();
+  TF_RETURN_IF_ERROR(AppendCommands(
+      *(cmd_sequence.get()), embed_thunk->thunks(),
```
Comment (ezhulenev) [inline xla/service/gpu/runtime/command_buffer_cmd_emitter.cc:198]: `*cmd_sequence` ?

File: xla/hlo/ir/hlo_schedule.h [hunk 3]
```diff
@@ -86,6 +86,11 @@ class HloInstructionSequence {
     id_sequence_.insert(id_sequence_.begin() + index, instruction->unique_id());
   }
 
+  bool contains(const HloInstruction* inst) const {
+    return std::find(instruction_sequence_.begin(), instruction_sequence_.end(),
```
Comment (ezhulenev) [inline xla/hlo/ir/hlo_schedule.h:90]: `absl::c_find`

File: xla/service/gpu/transforms/command_buffer_scheduling.cc (hunk 1 above)
Comment (shawnwang18) [reply xla/service/gpu/transforms/command_buffer_scheduling.cc:246]: Seems that the config name still has `address_computation` as name, we can have another PR to change the name?

File: xla/service/gpu/runtime/command_buffer_cmd_emitter.cc (hunk 2 above)
Comment (shawnwang18) [reply xla/service/gpu/runtime/command_buffer_cmd_emitter.cc]: Fixed

File: xla/hlo/ir/hlo_schedule.h (hunk 3 above)
Comment (shawnwang18) [reply xla/hlo/ir/hlo_schedule.h]: Fixed

**Review by ezhulenev — APPROVED** [review]

**Pull request closed without merging**